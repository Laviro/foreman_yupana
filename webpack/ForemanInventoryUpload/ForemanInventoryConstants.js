import { translate as __ } from 'foremanReact/common/I18n';

export const INVENTORY_PAGE_TITLE = __('Red Hat Inventory Uploads');
