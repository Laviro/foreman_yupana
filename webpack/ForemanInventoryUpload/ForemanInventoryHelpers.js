import { foremanUrl } from '../ForemanRhCloudHelpers';

export const inventoryUrl = path =>
  foremanUrl(`/foreman_inventory_upload/${path}`);

export const getInventoryDocsUrl = () =>
  'https://access.redhat.com/products/subscription-central';
