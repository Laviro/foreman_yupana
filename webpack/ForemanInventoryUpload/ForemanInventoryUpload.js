import React from 'react';
import { IntlProvider } from 'react-intl';
import { Grid } from 'patternfly-react';
import AccountList from './Components/AccountList';
import PageHeader from './Components/PageHeader';

const ForemanInventoryUpload = () => (
  <IntlProvider locale={navigator.language}>
    <Grid fluid className="inventory-upload">
      <PageHeader />
      <AccountList />
    </Grid>
  </IntlProvider>
);

export default ForemanInventoryUpload;
