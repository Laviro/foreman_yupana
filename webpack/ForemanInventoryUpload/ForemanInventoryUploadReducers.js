import { combineReducers } from 'redux';
import { reducers as accountListReducers } from './Components/AccountList';
import { reducers as dashboardReducers } from './Components/Dashboard';
import { reducers as filterReducers } from './Components/InventoryFilter';
import { reducers as inventorySyncReducers } from './Components/PageHeader/components/SyncButton';

export default {
  inventoryUpload: combineReducers({
    ...accountListReducers,
    ...dashboardReducers,
    ...filterReducers,
    ...inventorySyncReducers,
  }),
};
