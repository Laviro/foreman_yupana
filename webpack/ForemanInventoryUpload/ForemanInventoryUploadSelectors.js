// export const selectForemanInventoryUpload = state => state.ForemanInventoryUpload || {};
// Added 'state' to use redux in storybook only, should be removed.
export const selectForemanInventoryUpload = state => state.ForemanInventoryUpload || state || {};
