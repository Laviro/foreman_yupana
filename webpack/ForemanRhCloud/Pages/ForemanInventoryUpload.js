import React from 'react';
import { Grid } from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';
import AccountList from '../Components/AccountList';
import PageHeader from '../Components/PageHeader';
import PageLayout from '../../__mocks__/foremanReact/routes/common/PageLayout/PageLayout';

const ForemanInventoryUpload = () => (
  <PageLayout header={__('Red Hat Inventory Upload')} searchable={false}>
    <Grid fluid className="inventory-upload">
      <PageHeader />
      <AccountList />
    </Grid>
  </PageLayout>
);

export default ForemanInventoryUpload;
