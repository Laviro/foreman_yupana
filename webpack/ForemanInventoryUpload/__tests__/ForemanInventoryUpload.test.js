import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import ForemanInventoryUpload from '../../ForemanInventoryUpload';

const fixtures = {
  'render without Props': {},
};

describe('ForemanInventoryUpload', () =>
  testComponentSnapshotsWithFixtures(ForemanInventoryUpload, fixtures));
