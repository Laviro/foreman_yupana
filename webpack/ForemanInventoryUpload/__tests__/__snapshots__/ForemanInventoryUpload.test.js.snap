// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForemanInventoryUpload render without Props 1`] = `
<div
  className="rh-cloud-inventory-page"
>
  <PageLayout
    beforeToolbarComponent={<PageHeader />}
    searchable={false}
  >
    <Connect(AccountList) />
  </PageLayout>
</div>
`;
